src/plugins/data_source/server/saved_objects/data_source_saved_objects_client_wrapper.ts

```diff
@@ -0,0 +1,273 @@
+            return object;
+          }
+
+          const encryptedAttributes: Partial<T> = await this.validateAndUpdatePartialAttributes(
+            attributes
+          );
+
+          return {
+            ...object,
+            attributes: encryptedAttributes,
+          };
+        })
+      );
+
+      return await wrapperOptions.client.bulkUpdate(objects, options);
+    };
+
+    return {
+      ...wrapperOptions.client,
+      create: createWithCredentialsEncryption,
+      bulkCreate: bulkCreateWithCredentialsEncryption,
+      checkConflicts: wrapperOptions.client.checkConflicts,
+      delete: wrapperOptions.client.delete,
+      find: wrapperOptions.client.find,
+      bulkGet: wrapperOptions.client.bulkGet,
+      get: wrapperOptions.client.get,
+      update: updateWithCredentialsEncryption,
+      bulkUpdate: bulkUpdateWithCredentialsEncryption,
+      errors: wrapperOptions.client.errors,
+      addToNamespaces: wrapperOptions.client.addToNamespaces,
+      deleteFromNamespaces: wrapperOptions.client.deleteFromNamespaces,
+    };
+  };
+
+  private isValidUrl(endpoint: string) {
+    try {
+      return Boolean(new URL(endpoint));
+    } catch (e) {
+      return false;
+    }
+  }
+
+  private async validateAndEncryptAttributes<T = unknown>(attributes: T) {
+    this.validateAttributes(attributes);
+
+    const { auth } = attributes;
+
+    switch (auth.type) {
+      case AuthType.NoAuth:
+        return {
+          ...attributes,
+          // Drop the credentials attribute for no_auth
+          credentials: undefined,
+        };
+      case AuthType.UsernamePasswordType:
+        return {
+          ...attributes,
+          auth: await this.encryptCredentials(auth),
+        };
+      default:
+        throw SavedObjectsErrorHelpers.createBadRequestError(`Invalid auth type: '${type}'`);
+    }
+  }
+
+  private async validateAndUpdatePartialAttributes<T = unknown>(attributes: T) {
+    const { auth, endpoint } = attributes;
+
+    if (endpoint) {
+      throw SavedObjectsErrorHelpers.createBadRequestError(
+        `Update data source endpoint is not supported`
+      );
+    }
+
+    if (auth === undefined) {
+      return attributes;
+    }
+
+    const { type, credentials } = auth;
+
+    switch (type) {
+      case AuthType.NoAuth:
+        return {
+          ...attributes,
+          // Drop the credentials attribute for no_auth
+          credentials: undefined,
+        };
+      case AuthType.UsernamePasswordType:
+        if (credentials && credentials.password) {
+          return {
+            ...attributes,
+            auth: await this.encryptCredentials(auth),
+          };
+        } else {
+          return attributes;
+        }
+      default:
+        throw SavedObjectsErrorHelpers.createBadRequestError(`Invalid credentials type: '${type}'`);
+    }
+  }
+
+  private validateAttributes<T = unknown>(attributes: T) {
+    const { title, endpoint, auth } = attributes;
+    if (!title) {
```

Comment:
This is a good call. Just tested with creating visualizations though, the name could be just spaces. O_O.  But this will leave it in a broken state, so I say we shouldn't allow only spaces here.